packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx

```diff
@@ -0,0 +1,320 @@
+              </>
+            ) : nftState.buyNowPrice ? (
+              <>
+                <GridItem colSpan={buttonColumnSpan}>
+                  <ButtonGrid data-size={size} data-two-columns>
+                    <Button fullWidth variant="secondary" size={buttonSize}>
+                      Place a bid
+                    </Button>
+                    <Button fullWidth size={buttonSize}>
+                      Buy now
+                    </Button>
+                    {/* second row button */}
+                    {nftState.canWithdrawBid && (
+                      <GridItem colSpan={2}>
+                        <Button fullWidth size={buttonSize} variant="destructive-secondary">
+                          Withdraw a bid
+                        </Button>
+                      </GridItem>
+                    )}
+                  </ButtonGrid>
+                </GridItem>
+              </>
+            ) : (
+              <>
+                <GridItem colSpan={buttonColumnSpan}>
+                  <ButtonGrid data-size={size}>
+                    <Button fullWidth size={buttonSize}>
+                      Place a bid
+                    </Button>
+                    <Button fullWidth size={buttonSize} variant="destructive-secondary">
+                      Withdraw a bid
+                    </Button>
+                  </ButtonGrid>
+                </GridItem>
+              </>
+            )}
+          </>
+        )
+    }
+  }, [size, nftState, isOwner])
+
+  return (
+    <Container ref={containerRef}>
+      <NFTOwnerContainer data-size={size}>
+        <OwnerAvatar assetUrl={ownerAvatarUri} size="small" />
+        <OwnerLabel variant="t100" secondary>
+          This NFT is owned by
+        </OwnerLabel>
+        <OwnerHandle to={ownerHandle && absoluteRoutes.viewer.member(ownerHandle)} variant="secondary" textOnly>
+          {ownerHandle}
+        </OwnerHandle>
+      </NFTOwnerContainer>
+      <Content data-size={size}>{content}</Content>
+
+      {/* TODO: add history */}
+      <NFTHistoryHeader data-size={size}>
+        <Text variant={size === 'small' ? 'h300' : 'h400'}>History</Text>
+        <SvgActionChevronB />
+      </NFTHistoryHeader>
+    </Container>
+  )
+}
+
+type NFTInfoItemProps = { size: Size; label: string; content: React.ReactNode; secondaryText?: string }
+const NFTInfoItem: React.FC<NFTInfoItemProps> = ({ size, label, content, secondaryText }) => {
+  return (
+    <InfoItemContainer data-size={size}>
+      <Label variant="h100" secondary>
+        {label}
+      </Label>
+      <InfoItemContent data-size={size}>{content}</InfoItemContent>
+      <SecondaryText as="p" variant="t100" secondary data-size={size}>
+        {secondaryText}
+      </SecondaryText>
+    </InfoItemContainer>
+  )
+}
+
+const NFTTimerItem: React.FC<{ size: Size; time: Date }> = ({ size, time }) => {
+  const [, rerender] = React.useState({})
+  const forceUpdate = React.useCallback(() => rerender({}), [])
```

Comment:
this looks very hacky to me. Why we can just set a time left every second inside useEffect?

Reply:
I don't think there is anything too unusual going on here, I liked it a bit more coz I didn't have to care about handling the first value of the useState; made it into a useState now.